lib/transactions.rb

```diff
@@ -0,0 +1,41 @@
+class Transactions
+
+  attr_reader :list
+
+  def initialize
+    @list = []
+  end
+
+  def store_transaction(amount, balance)
+    self.list.push({date: format_date, amount: amount, balance: balance})
+  end
+
+  def format_header
+    puts "Date || Credit || Debit || Balance"
+  end
+
+  def format_statement
+    list.reverse.each do |item|
+      format_amount
+      puts "#{item[:date]} || #{item[:credit]} || #{item[:debit]} || #{item[:balance]}"
+    end
+  end
+
+  def format_amount
+    list.each do |item|
+      if item.has_key?(:debit) || item.has_key?(:credit)
+        item
+      elsif item[:amount] < 0
+        item[:debit] = item.delete :amount
```

Comment:
Is it `format_amount`'s responsibility to work out if a transaction is a credit or debit? This breaks the encapsulation of the transaction object.